packages/salesforcedx-vscode-lwc/src/dxsupport/waitForDX.ts

```diff
@@ -0,0 +1,40 @@
+import * as vscode from 'vscode';
+
+let wait: Promise<vscode.Extension<any>>;
+
+export async function waitForDX(activate: boolean = false) {
```

Comment:
This is unreliable since activation times vary based on the machine's resources. Why do we want to activate and then wait for `salesforcedx-vscode-core` to be present instead of having a direct dependency with `salesforcedx-vscode-core`? Doing so will trigger activation of this extension after core is active.

Reply:
There isn't much that we use from the salesforcedx-vscode-core plugin (the only thing we need is the TelemetryService and the createLwcComponent command, both of which the language server can function without). Avoiding the direct dependency allows us to do things immediately (like indexing / content assist, for example). Otherwise we have to wait until core is loaded before the language server can do anything.

Reply:
How big of a sticking point is this? It seems like an optimization, but if you have a use case where you think this would cause bugs, let me know

Reply:
This has two purposes: 1) improve startup performance, 2) make the dependency optional. 

If we have a direct dependency, dx will have to active before we start to activate, slowing down the entire activation process. 

In a core scenario (working on lightning/lwc on a salesforce core workspace) we do not need the vscode-core dependency at all. It would be wrong, actually, since the lightning commands that we use from the toolbelt (like create lightning component) wouldn't work in a core env. 

We can increase the length of time we wait for DX to start. Hopefully it starts within 2 minutes, but we can make it 5 or 10 or whatever you think is appropriate.

Reply:
Regarding the startup time, the release that just went out today pretty much eliminated any startup time - Luis basically removed everything from the activation and loaded everything async. 

Regarding the optional dependency on core, I'm wondering if we can still require it to be installed but basically, it won't do anything - the core extension shouldn't do anything unless it detects an `sfdx-project.json` so it wouldn't add any functionality for core devs. Correct me if I'm wrong luis.

Reply:
Looks like the real concern for adding this was performance. Has this been tested against the latest changes in `develop` branch ? This week's changes significantly improved the performance of `salesforcedx-vscode-core` activation/start up.

Reply:
I also do agree that having a direct dependency in core just because of the telemetry functionality is a bit too much. Might be a good time to start exploring moving that into its own utility module.